python/tank/platform/software_launcher.py

```diff
@@ -336,47 +375,83 @@ def get_standard_plugin_environment(self):
 
         return env
 
-    def is_version_supported(self, version):
+    def is_version_supported(self, sw_version):
```

Comment:
It feels overkill that we have to construct a `SoftwareVersion` object just to compare it. This feels like it limits the way this method could be utilised by calling classes. Also confusing that it's called `is_version_supported` when it checks both version and product. How about something like

```python
def _is_supported(self, version, product):
    """
    Helper method that can be used by implementing classes to check
    agasinst built-in constraints.

    :param str version: Version number string
    :param str product: Product string
    """
```

Seems better that the method is protected if it's meant to be used as a helper for deriving classes.